routes/project/location/postgres.rb

```diff
@@ -308,6 +308,87 @@ class Clover
         response.headers["Content-Type"] = "application/x-pem-file"
         certs
       end
+
+      r.get "metrics" do
+        authorize("Postgres:view", pg.id)
+
+        start_ts = request.params["start"]&.to_i || Time.now.utc.to_i - 30 * 60
+        end_ts = request.params["end"]&.to_i || Time.now.utc.to_i
+
+        if end_ts < start_ts
+          raise CloverError.new(400, "InvalidRequest", "End timestamp must be greater than start timestamp")
+        end
+
+        if end_ts - start_ts > 31 * 24 * 60 * 60 + 5 * 60
+          raise CloverError.new(400, "InvalidRequest", "Maximum time range is 31 days")
+        end
+
+        if start_ts < Time.now.utc.to_i - 31 * 24 * 60 * 60
+          raise CloverError.new(400, "InvalidRequest", "Cannot query metrics older than 31 days")
+        end
+
+        metric_key = request.params["key"]&.to_sym
+        single_query = !request.params["key"].nil?
+
+        if single_query && !Metrics::POSTGRES_METRICS.key?(metric_key)
+          raise CloverError.new(400, "InvalidRequest", "Invalid metric name")
+        end
+
+        metric_keys = metric_key ? [metric_key] : Metrics::POSTGRES_METRICS.keys
+
+        vmr = VictoriaMetricsResource.first(project_id: Config.victoria_metrics_service_project_id)
```

Comment:
Shouldn't project_id of the VictoriaMetricsResource be `Config.postgres_service_project_id` (i.e. VictoriaMetricsResource belongs to Postgres)?

project_id of the VMs should be `Config.victoria_metrics_service_project_id` (i.e. VMs belong to VictoriaMetrics)

Reply:
I think this makes more sense. There are a couple of other parts that assume otherwise, let me update those here.